### PR TITLE
ref(ts): Convert accountSecurityDetails to typescript

### PR DESCRIPTION
**Pull request by priscilawebdev**
Convert accountSecurityDetails to typescript

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx [hunk 1]
```diff
@@ -21,91 +21,86 @@ import TextBlock from 'app/views/settings/components/text/textBlock';
...
-    );
+
+    // if the device is defined, it means that U2f is being removed
+    const deviceId = device ? `${device.key_handle}/` : '';
+    const deviceName = device ? device.name : 'Authenticator';
+
+    this.setState({loading: true});
+
+    try {
+      await this.api.requestPromise(`${ENDPOINT}${authenticator.authId}/${deviceId}`, {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx:81]: I'm not sure if we should make this request if `deviceId` is an empty string, but I think it's better not to change this logic now.

File: src/sentry/static/sentry/app/types/index.tsx [hunk 2]
```diff
@@ -606,6 +606,12 @@ export type GlobalSelection = {
   };
 };
 
+type Device = {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/types/index.tsx:609]: IMO not a great type name to have in `app/types`. Maybe `AuthenticatorDevice`

File: src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx [hunk 3]
```diff
@@ -21,91 +21,86 @@ import TextBlock from 'app/views/settings/components/text/textBlock';
...
+      // Error deleting authenticator
+      this.setState({loading: false});
+      addErrorMessage(t('Error removing %s', deviceName));
+    }
   };
 
   renderBody() {
     const {authenticator} = this.state;
+
+    if (!authenticator) {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx:96]: This is what i meant about adding `null` to `AsyncComponent` things, `renderBody` will never render when state from requests is null

Comment (priscilawebdev) [reply src/sentry/static/sentry/app/views/settings/account/accountSecurity/accountSecurityDetails.tsx:96]: that's true, but I prefer to leave it at that for now...let's discuss this topic in the upcoming TSC meeting 😉

**Review by markstory — APPROVED** [review]

**Pull request merged**